api/scorings.go

```diff
@@ -0,0 +1,124 @@
+package api
+
+import "math"
+
+// ThresholdRange represents the lowest, highest value and score of each range of threshold.
+type ThresholdRange struct {
+	Min   float64
+	Max   float64
+	Score float64
+}
+
+// ScoringRules reprepsents threshold range and weigh of each metrics.
+type ScoringRule struct {
+	Ranges []ThresholdRange
+	Weight float64
+}
+
+func calculateScore(value float64, rule ScoringRule) float64 {
+	for _, r := range rule.Ranges {
+		if value >= r.Min && value < r.Max {
+			return float64(r.Score) * rule.Weight
+		}
+	}
+	return 0
+}
```

Comment:
* This doesn't calculate, it applies weighted factors. I would rename to `weigh` or `applyFactors`
* Usually it's better to have the nested/if statements in a loop be for exclusion, i.e. what we don't care about, and the actual body of the loop to be the thing we care about. So in this case we want to skip the values that arent within the threshold, and return when we find one that is within the threshold. WDYT?

```suggestion
func applyFactors(value float64, rule ScoringRule) float64 {
	for _, r := range rule.Ranges {
		if value < r.Min || value >= r.Max {
			continue
		}
		
		return float64(r.Score) * rule.Weight
	}
	return 0
}
```

Reply:
Oh I see 🤩, it's cleaner and more readable. Love it !